File: js/tasks/ransom-note.js
'use strict';

process.stdin.resume();
process.stdin.setEncoding('utf-8');

let inputString = '';
let currentLine = 0;

process.stdin.on('data', inputStdin => {
    inputString += inputStdin;
});

process.stdin.on('end', function() {
    inputString = inputString.replace(/\s*$/, '')
        .split('\n')
        .map(str => str.replace(/\s*$/, ''));

    main();
});

function readLine() {
    return inputString[currentLine++];
}

// Complete the checkMagazine function below.
function checkMagazine(magazine, note) {
    let scraps = {};
    magazine.forEach((m) => {
        scraps[m] = scraps[m] ? scraps[m] + 1 : 1;
    });

    # avoid forEach to avoid async closure
    for (let n = 0; n < note.length; n++) {
        let word = note[n];
        if (scraps[word] && scraps[word] > 0) {
            scraps[word] -= 1;
        } else {
            console.log('No');
            return
        }
    };
    console.log('Yes');
}

function main() {
    const mn = readLine().split(' ');

    const m = parseInt(mn[0], 10);

    const n = parseInt(mn[1], 10);

    const magazine = readLine().split(' ');

    const note = readLine().split(' ');

    checkMagazine(magazine, note);
}

File: python/tasks/ransom-note.py
#!/bin/python

import math
import os
import random
import re
import sys

# Complete the checkMagazine function below.
def checkMagazine(magazine, note):
    from collections import Counter
    print 'Yes' if ((Counter(note) - Counter(magazine)) == {}) else 'No'

if __name__ == '__main__':
    mn = raw_input().split()

    m = int(mn[0])

    n = int(mn[1])

    magazine = raw_input().rstrip().split()

    note = raw_input().rstrip().split()

    checkMagazine(magazine, note)

